pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -1660,3 +1660,34 @@ func MutateShootETCDEncryptionKeyRotation(shoot *gardencorev1beta1.Shoot, f func
 
 	f(shoot.Status.Credentials.Rotation.ETCDEncryptionKey)
 }
+
+// IsValidShootOperation checks whether the provided operation annotation value is valid. It returns two bools: The
+// first value specifies whether it's valid, and the second bool value specifies whether the gardener-apiserver removes
+// the annotation before the Shoot is persisted to etcd.
+func IsValidShootOperation(operation string, lastOperation *gardencorev1beta1.LastOperation) (isValid bool, removedBeforePersisted bool) {
+	if lastOperation != nil && lastOperation.State == gardencorev1beta1.LastOperationStateFailed {
+		if operation == v1beta1constants.ShootOperationRetry {
+			return true, true
+		}
+	} else {
+		switch operation {
+		case v1beta1constants.GardenerOperationReconcile:
+			return true, true
+
+			// TODO(rfranzke): Handle new operation annotations introduced by
+			// - https://github.com/gardener/gardener/pull/6021
+			// - https://github.com/gardener/gardener/pull/6038
+			// once those PRs got merged.
+		case v1beta1constants.ShootOperationRotateKubeconfigCredentials,
+			v1beta1constants.ShootOperationRotateObservabilityCredentials,
+			v1beta1constants.ShootOperationRotateSSHKeypair,
+			v1beta1constants.ShootOperationRotateCAStart,
+			v1beta1constants.ShootOperationRotateCAComplete,
+			v1beta1constants.ShootOperationRotateServiceAccountKeyStart,
+			v1beta1constants.ShootOperationRotateServiceAccountKeyComplete:
+			return true, false
+		}
```

Comment:
@oliver-goetz had the suggestion to have a validation on apiserver side for the valid/supported operation annotation values. From user's perspective you are now allowed to add operation annotation value that is not valid.

Reply:
Good point, I think it makes sense given hat the operation annotations are quite important. However, I think it makes sense to tackle this outside of this PR. @oliver-goetz could you please create an issue for the topic?